src/material/datepicker/month-view.ts

```diff
@@ -285,9 +288,14 @@ export class MatMonthView<D> implements AfterContentInit, OnChanges, OnDestroy {
         break;
       case ENTER:
       case SPACE:
-        if (!this.dateFilter || this.dateFilter(this._activeDate)) {
-          this._dateSelected({value: this._dateAdapter.getDate(this._activeDate), event});
+        this._selectionKeyPressed = true;
```

Comment:
Do we need the `_selectionKeyPressed` stuff? I would think if we get a `keyup` for `ENTER`/`SPACE` it's safe to assume there was a `keydown`

Reply:
The keydown could've come from outside the calendar. E.g. the browser fires the `click` when doing an `ENTER` keydown on the datepicker toggle.